subprojects/docs/src/samples/userguide/artifacts/defineRepository/kotlin/build.gradle.kts

```diff
@@ -0,0 +1,305 @@
+// tag::maven-ivy-repository-no-auth[]
+repositories {
+    maven {
+        setUrl("http://repo.mycompany.com/maven2")
+    }
+
+    ivy {
+        setUrl("http://repo.mycompany.com/repo")
+    }
+}
+// end::maven-ivy-repository-no-auth[]
+
+// tag::maven-ivy-repository-auth[]
+repositories {
+    maven {
+        setUrl("sftp://repo.mycompany.com:22/maven2")
+        credentials {
+            username = "user"
+            password = "password"
+        }
+    }
+
+    ivy {
+        setUrl("sftp://repo.mycompany.com:22/repo")
+        credentials {
+            username = "user"
+            password = "password"
+        }
+    }
+}
+// end::maven-ivy-repository-auth[]
+
+// tag::maven-ivy-s3-repository[]
+repositories {
+    maven {
+        setUrl("s3://myCompanyBucket/maven2")
+        credentials(AwsCredentials::class.java) {
```

Comment:
Is using `.java` still needed? 🤔 
I thought with 4.10-rcX it is possible to use the Kotlin classes?

Reply:
Indeed, `AwsCredentials::class` works fine. Unfortunately, this is yet another case where The build is successful, but IntelliJ marks the code as red (`Type mismatch. Required: Class<???..???> Found KClass<AwsCredentials>`).
I'll leave this as is for now, until a core member notices the issue and advises.